### DISCUSSION TITLE
Setting up with single type

### DISCUSSION THREAD

**Discussion by Darkmagister** [strapi-community/plugin-io#89] [Q&A]
i'm using a single type in strapi to save some configurations, and i would like that when someone in strapi update those config to emit an event via socketio so that i can get the new values, i've installed the plugin and configure like this:
```
'io': {
        enabled: true,
        config: {
            socket: {
                serverOptions: {
                    cors: { origin: '*', methods: ['GET', 'POST'] }
                }
            },
            contentTypes: ['api::setting.setting'],
            events: [
                {
                    name: 'connection',
                    handler({ strapi, io }, socket) {
                        // will log whenever a socket connects
                        strapi.log.info(`[io] new connection with id ${socket.id}`)
                    }
                }
            ]
        }
    }
```

then in the lifecycles.js file of the single type i added this:

```
afterUpdate(event) {
        const { result, params } = event
        console.log('afterUpdate', result)
        strapi.$io.emit('api::setting.setting.find', result)
    }
```
i used find as event here as setting.find is the only permission available to public (no auth needed for this)

for the client i will be using vue but for testing now i'm trying this:
https://addons.mozilla.org/en-US/firefox/addon/socketio-client/

when i connect i get the console with the id, but when i update the setting in strapi i get the console.log but no event on the client can anyone have any suggestion?

thanks

**Comment by ComfortablyCoding** [comment]
You should not need to emit the data manually for any content types defined in the plugin config. If you want to listen for setting updates all that should be needed is the listener on the client side.

```js
socket.on('connect', () => {
	socket.on('setting:update', (data) => {
		console.log('settings updated!');
		console.log(data);
	});
});
```

In terms of the find action not working, it is not a supported action at this time. You can utilize the raw emit function if you still wish to emit it. Keep in mind that data will not be transformed in raw and will need to be done manually as well.

**Reply by Darkmagister** [reply]
ok, if i add the update permission to public i receive the event setting:update, but ... i don't want to add update permission, as i just need notification on the update but i don't want to be able to update it from the role.

how can i emit an update event even without the update permission? this is why i have used a custom emit that use the setting:find so the find should be the one that need to be enabled in the permission  but with the find i don't get any event.

**Reply by ComfortablyCoding** [reply]
As of now this is only possible via the raw emit function.

I will look into adding support for the find operation as well.
